include/onnxruntime/interface/graph/graph.h

```diff
@@ -0,0 +1,190 @@
+#ifdef INTREE_EP
+#include "onnx/onnx_pb.h"
+#endif
+
+namespace onnxruntime {
+/// <summary>
+/// Enum of DataTypes using standard ONNX values. Casting to/from int32_t is encouraged.
+/// </summary>
+enum class DataType : int32_t {
+  UNDEFINED = 0,
+  FLOAT = 1,   // float
+  UINT8 = 2,   // uint8_t
+  INT8 = 3,    // int8_t
+  UINT16 = 4,  // uint16_t
+  INT16 = 5,   // int16_t
+  INT32 = 6,   // int32_t
+  INT64 = 7,   // int64_t
+  STRING = 8,  // string
+  BOOL = 9,    // bool
+  FLOAT16 = 10,
+  DOUBLE = 11,
+  UINT32 = 12,
+  UINT64 = 13,
+  COMPLEX64 = 14,
+  COMPLEX128 = 15,
+  BFLOAT16 = 16,
+  FLOAT8E4M3FN = 17,
+  FLOAT8E4M3FNUZ = 18,
+  FLOAT8E5M2 = 19,
+  FLOAT8E5M2FNUZ = 20,
+};
+
+namespace interface {
+class GraphViewRef;
+/// <summary>
+/// An interface for a constant tensor value used by initializers
+/// </summary>
+class TensorRef {
+ public:
+  /// <returns>The shape of the tensor. Values are nonnegative.</returns>
+  virtual std::vector<int64_t> Shape() const = 0;
+
+  virtual size_t NumElements() const = 0;
+
+  /// <returns>The dtype of the tensor.</returns>
+  virtual DataType DType() const = 0;
+
+  /// <summary>
+  /// Retrieves copy of raw data bytes from the tensor. Used for reading initializers specifying axes/pads/scales.
+  /// </summary>
+  /// <returns>Flattened tensor data in bytes</returns>
+  virtual std::vector<uint8_t> Data() const = 0;
+
+  virtual ~TensorRef(){};
+};
+
+class ValueInfoViewRef {
+ public:
+   /// <returns>The name of the value in the graph</returns>
+  virtual std::string_view Name() const = 0;
+
+  /// <returns>
+  /// The inferred/declared tensor shape of the value. nullopt if rank is unknown, otherwise a vector with entries
+  /// representing the dimensions of the value. Use -1 for unknown dimensions.
+  /// </returns>
+  virtual std::optional<std::vector<int64_t>> Shape() const = 0;
+
+  /// <returns>The inferred/declared dtype of the value. UNDEFINED (0) if dtype is unknown.</returns>
+  virtual DataType DType() const = 0;
+
+  virtual ~ValueInfoViewRef(){};
+};
+
+class NodeViewRef {
+ public:
+  virtual size_t Index() const = 0;
+
+  /// <returns>Op computed by the node</returns>
+  virtual std::string_view OpType() const = 0;
+
+  /// <returns>Domain containing the op. Empty string if node has no domain set.</returns>
+  virtual std::string_view Domain() const = 0;
+
+  /// <returns>Names of input values. Empty string may be included for optional inputs.</returns>
+  virtual std::vector<std::string_view> Inputs() const = 0;
+
+  /// <returns>Names of output values. Empty string may be included for optional outputs.</returns>
+  virtual std::vector<std::string_view> Outputs() const = 0;
+
+  /// <param name="name">Name of the attribute to return</param>
+  /// <returns>
+  /// The attribute value, or nullopt if the attribute is not present on the node, or is not of type int.
+  /// </returns>
+  virtual std::optional<int64_t> GetAttributeInt(std::string_view name) const = 0;
```

Comment:
>GetAttributeInt(std::string_view name) const = 0; [](http://example.com/codeflow?start=33&length=49)

you will need to support other attribute types, like floats, right?

Reply:
Yes, already added floats, need to add float as well